### PR TITLE
Feature/feature unsubscribe group

### PR DESCRIPTION
**Pull request by xavier-campos-diaz**

### CODE REVIEW

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/InfoGroupSubscriptionsFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteSubscription.java  7
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeleteSubscription.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5389854&bid=17670248)

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteSubscription.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * @author Xavier Campos
+ */
+public class TrDeleteSubscription {
+    private CommunityService communityService;
+    private User user;
+    private Group group;
+
+    public TrDeleteSubscription(CommunityService communityService) {
```
Comment (AlbertPG) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteSubscription.java:18]: You should comment this class as well

File: app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeleteSubscription.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ */
+public class TestTrDeleteSubscription {
+    private User user;
+    private Group group;
+    private TrDeleteSubscription trDeleteSubscription;
+
+    @Before
+    public void setUp() {
+        user = new User("John Smith", "johnSmith@gmail.com", "1234");
+        group = new Group("Husky", "John Doe", DateTime.Builder.buildDateString("2020-04-15"));
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeleteSubscription.java:26]: Codacy Issue found: [The String "2020-04-15" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5389854)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/community/Group.java  8
         

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/InfoGroupSubscriptionsFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteSubscription.java  7
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeleteSubscription.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5389854&bid=17670248)

File: app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeleteSubscription.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        trDeleteSubscription.setUser(user);
+        trDeleteSubscription.setGroup(new Group("Penguins", "Oriol Simo",
+            DateTime.Builder.buildDateString("2020-04-15")));
+        trDeleteSubscription.execute();
+    }
+
+    @Test(expected = NotSubscribedException.class)
+    public void shouldNotDeleteSubscriptionFromNotSubscribedGroup() throws GroupNotExistingException,
+        NotSubscribedException, OwnerCannotDeleteSubscriptionException {
+        StubCommunityService communityService = new StubCommunityService();
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeleteSubscription.java:42]: Codacy Issue found: [Avoid unused local variables such as 'communityService'.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5389854)

**Review by AlbertPG — APPROVED** [review]
After commenting this class, everything is correct

**Review by AlbertPG — APPROVED** [review]
Okey, I approve it

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/community/Group.java  8
         

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/InfoGroupSubscriptionsFragment.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeleteSubscription.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeleteSubscription.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5389854&bid=17670248)

**Pull request merged**